Mage.Sets/src/mage/sets/antiquities/Powerleech.java

```diff
@@ -0,0 +1,97 @@
+/*
+ *  Copyright 2010 BetaSteward_at_googlemail.com. All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without modification, are
+ *  permitted provided that the following conditions are met:
+ *
+ *     1. Redistributions of source code must retain the above copyright notice, this list of
+ *        conditions and the following disclaimer.
+ *
+ *     2. Redistributions in binary form must reproduce the above copyright notice, this list
+ *        of conditions and the following disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY BetaSteward_at_googlemail.com ``AS IS'' AND ANY EXPRESS OR IMPLIED
+ *  WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ *  FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL BetaSteward_at_googlemail.com OR
+ *  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ *  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ *  ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ *  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ *  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *  The views and conclusions contained in the software and documentation are those of the
+ *  authors and should not be interpreted as representing official policies, either expressed
+ *  or implied, of BetaSteward_at_googlemail.com.
+ */
+package mage.sets.antiquities;
+
+import java.util.UUID;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.effects.common.GainLifeEffect;
+import mage.cards.CardImpl;
+import mage.constants.CardType;
+import mage.constants.Rarity;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+
+/**
+ *
+ * @author MarcoMarin
+ */
+public class Powerleech extends CardImpl {
+
+    public Powerleech(UUID ownerId) {
+        super(ownerId, 64, "Powerleech", Rarity.UNCOMMON, new CardType[]{CardType.ENCHANTMENT}, "{G}{G}");
+        this.expansionSetCode = "ATQ";
+
+        // Whenever an artifact an opponent controls becomes tapped or an opponent activates an artifact's ability without {tap} in its activation cost, you gain 1 life.
+        this.addAbility(new AbilityActivatedTriggeredAbility3());
+    }
+
+    public Powerleech(final Powerleech card) {
+        super(card);
+    }
+
+    @Override
+    public Powerleech copy() {
+        return new Powerleech(this);
+    }
+}
+class AbilityActivatedTriggeredAbility3 extends TriggeredAbilityImpl {

```

Comment:
Ok.. the comment I referred to alluded to the effect, which is a different topic. (still would like to know that tho :))

About the reusable class, one must be careful because Powerleech has the additional clause to check for Opponents. (which I'd be grateful if you'd tell me how to check it properly, something like Target.Opponent? My way only works on 1v1 games)

About the "differentiating between artifact activated abilities with tap or not." We need to think a bit through here I guess. So, we agree at least tapping would work (2nd clause). Right?

So the problem is "EventType.ACTIVATED_ABILITY" you're saying applies to all of them (thus "not differentiating")?

 Ok, if we're still on the same page up to here, then I'd say this would work UNLESS there is an effect that allows a tap ability to be activated without actually tapping it and without removing the property of the ability having "tap" on it. (which sounds somewhat contrived :)) Because a normal (with tap) ability will fall in the 2nd case (artifact is tapped also triggers this effect).

If that is still a problem, then I think the solution would be to add the 2nd clause inverted to the first one, between parens. I.e. (A && !B) || B

 This is another reason I'm reluctant to add to the communal pool of effects outside of cards because I'm still too ignorant to code enough generality in them to not have them mislead the programmers. =/